query_optimizer/strategy/Join.cpp

```diff
@@ -371,6 +373,7 @@ void Join::addHashJoin(const logical::ProjectPtr &logical_project,
                           left_join_attributes,
                           right_join_attributes,
                           residual_predicate,
+                          build_predicate,
```

Comment:
Optionally, I think `E::PredicatePtr()  /* build_predicate */,` is better.

Reply:
Agreed, less superfluous that way.